assignment3/solve.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: UTF-8 -*-
+
+import unittest
+
+
+class BorNode(object):
+    def __init__(self):
+        self.edges = {}
+        self.word_end = False
+
+
+class Bor(object):
+    """
+    For solving this problem I used Bor data structure.
```

Comment:
Is it a data structure you invented? If not, can you please reference some article describing its properties?